### PR TITLE
Rename  'gcloud/_apitools' -> 'gcloud/streaming'

### PR DESCRIPTION
**Pull request by tseaver**
Uses #1207 as a base.

Works around a bug in 'nose', which forcibly ignores subpackages with leading '_'.

See:
http://stackoverflow.com/questions/23615924
https://github.com/nose-devs/nose/blob/master/nose/loader.py#L169.

### CODE REVIEW

**Comment by dhermes** [comment]
Why `streaming`?

**Comment by tseaver** [comment]
"Streaming" is all we're using the code for.

**Comment by dhermes** [comment]
OK

**Comment by tseaver** [comment]
@dhermes rebased after merge of #1207.

**Comment by tseaver** [comment]
@dhermes do we mean to have AppVeyor wired in as a "must pass" at this point?

**Comment by dhermes** [comment]
We can ignore. Once I'm done tinkering, I plan to make it run on tagged commits only. Though I'm open to using it more than on just tagged commits if people like.

File: gcloud/streaming/test_transfer.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def test_FromFile_w_existing_file_w_override_wo_auto_transfer(self):
             self.assertFalse(download.auto_transfer)
             del download  # closes stream
             with open(filename, 'rb') as fileobj:
-                self.assertEqual(fileobj.read(), '')
```
Comment (dhermes) [inline gcloud/streaming/test_transfer.py]: What happened here to require this change? This is not like the others.

File: gcloud/streaming/test_transfer.py [hunk 2]
```diff
@@ -565,8 +565,8 @@ def test_GetRange_not_initialized(self):
     def test_GetRange_wo_total_size_complete(self):
         from six.moves import http_client
         from gcloud._testing import _Monkey
-        from gcloud._apitools import transfer as MUT
-        CONTENT = 'ABCDEFGHIJ'
+        from gcloud.streaming import transfer as MUT
+        CONTENT = b'ABCDEFGHIJ'
```
Comment (dhermes) [inline gcloud/streaming/test_transfer.py]: I'm seeing maybe the `b''` before wasn't so out of place, but still am curious as to the why.

**Comment by dhermes** [comment]
LGTM other than the questions about using byte strings explicitly.

File: gcloud/streaming/test_transfer.py (hunk 1 above)
Comment (tseaver) [inline gcloud/streaming/test_transfer.py]: That change came in via 1378947, "Fix Py3k testing glitches".  Until the renaming, the tests weren't being found by Travis-tox, and therefore some Py3k issues remained.

Comment (dhermes) [inline gcloud/streaming/test_transfer.py]: Ha. Staring me right in the face. Thanks.

**Comment by tseaver** [comment]
Explicit byte strings are required for Py3k compatibility: streaming upload / download payloads really *are* bytes.

**Pull request merged**